### PR TITLE
update the description list component in the ds it is no longer in alphabetical order

### PR DESCRIPTION
**Pull request by alessioventuriniAND**
### What is the context of this PR?

This fixes:#3000

We have updated the onsMetadata name to onsDescriptionList. This has been done by still supporting the old version of this component so no breaking change is introduced. 

### How to review this PR

Yarn start this branch and confirm that the new component works correctly. 

### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

**Review by rmccar — COMMENTED** [review]
I think its a bit confusing where its now called description in some places and description list others. It should be description list everywhere. As for the css the class could be `.ons-description-list` and then `&__items` instead of `.ons-description` and `&__list`. Just because description could mean something else

Alessio : Good point! I have now updated this :)

File: src/components/description-list/_macro.spec.js [hunk 1]
```diff
@@ -61,25 +61,25 @@ const EXAMPLE_METADATA_MINIMAL = {
...
     const results = await axe($.html());
     expect(results).toHaveNoViolations();
   });
 
   it('has the provided `id` attribute', () => {
     const $ = cheerio.load(
-      renderComponent('metadata', {
-        ...EXAMPLE_METADATA_MINIMAL,
+      renderComponent('description-list', {
+        ...EXAMPLE_DESCRIPTION_MINIMAL,
```
Comment (rmccar) [inline src/components/description-list/_macro.spec.js:82]:
```suggestion
        ...EXAMPLE_DESCRIPTION_LIST_MINIMAL,
```

**Review by rmccar — COMMENTED** [review]
Theres these plus a few more that should change I think

File: src/patterns/send-and-reply-to-messages/example-conversation.njk [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 <div class="ons-grid">
     <div class="ons-grid__col ons-col-8@m">
         <section role="region" aria-label="Conversation: BRES 2017 clarification required for Bolts &amp; Ratchets Ltd">
             <h1 class="ons-u-fs-l"><span class="ons-u-vh">Subject: </span>BRES 2017 clarification required for Bolts &amp; Ratchets Ltd</h1>
             {{
-                onsMetadata({
-                    "metadataLabel": "Information about the survey and respondent",
+                onsDescriptionList({
+                    "descriptionLabel": "Information about the survey and respondent",
```
Comment (rmccar) [inline src/patterns/send-and-reply-to-messages/example-conversation.njk:11]:
```suggestion
                    "descriptionListLabel": "Information about the survey and respondent",
```

File: src/components/description-list/_macro-options.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+| ---------------- | ------------- | -------- | -------------------------------------------------------------------------------------------------------------- |
+| id               | string        | false    | The HTML `id` for the description list (`<dl>`)                                                                |
+| classes          | string        | false    | Classes to apply to description list (`<dl>`)                                                                  |
+| descriptionListLabel | string        | false    | Sets the HTML `title` and `aria-label` attributes to provide additional information about the description list |
+| termCol          | number        | true     | The number of grid columns used for the `<dt>` elements above medium breakpoint                                |
+| descriptionCol   | number        | true     | The number of grid columns used for the `<dd>` elements above medium breakpoint                                |
+| itemsList        | array`<Item>` | true     | Settings for the terms and descriptions of the [description list items](#item)                                 |
```
Comment (rmccar) [inline src/components/description-list/_macro-options.md:8]: Formatting on this table needs updating

Comment (alessioventuriniAND) [reply src/components/description-list/_macro-options.md]: All fixed :)

File: src/components/description-list/example-description.njk [hunk 4]
```diff
@@ -1,8 +1,8 @@
-{% from "components/metadata/_macro.njk" import onsMetadata %}
+{% from "components/description-list/_macro.njk" import onsDescriptionList %}
```
Comment (rmccar) [inline src/components/description-list/example-description.njk:1]: This file should be called `example-description-list.njk`

**Review by rmccar — APPROVED** [review]

**Review by SriHV — APPROVED** [review]

**Pull request merged**